toxcore/crypto_core.h

```diff
@@ -147,6 +157,18 @@ void crypto_sha256(uint8_t *hash, const uint8_t *data, size_t length);
 non_null()
 void crypto_sha512(uint8_t *hash, const uint8_t *data, size_t length);
 
+/**
+ * @brief Compute an HMAC authenticator (32 bytes).
```

Comment:
Can you document the parameters? What kind of key is `key`? Where do you get one from?

What is `auth`? Most of the time, given a parameter like `public_key`, you can find a `PUBLIC_KEY_SIZE` somewhere nearby, but there's no `AUTH_SIZE` (there is a `HMAC_SIZE`, maybe this should be called `hmac`?). Better yet: put `uint8_t hmac[CRYPTO_HMAC_SIZE]`, or if you want you can keep the name `auth` with explicit size.